### ISSUE TITLE
SQLite and Decimal (converting to float)

### ISSUE THREAD

**Issue by turicas** [csingley/ofxtools#7]
I'm trying to save `stmtrs.ofx` into SQLite and getting this warning:

```
[...]/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:565:
SAWarning: Dialect sqlite+pysqlite does *not* support Decimal objects natively,
and SQLAlchemy must convert from floating point - rounding errors and other issues
may occur. Please consider storing Decimal numbers as strings or integers on this
platform for lossless storage.
```

`ofxalchemy` may convert `Decimal` automatically to `float` or `str` if the destination database is SQLite.

**Comment by csingley** [comment]
TL;DR executive summary:  if you care about penny rounding errors, use a real database.

Originally ofxalchemy.types.OFXNumeric stored data as string on the SQLite backend, but I abandoned that because it conflicted with setting SQL CHECK CONSTRAINT statements - most importantly to ensure that INVTRAN aggregates actually do have TOTAL set to the correct value so you can just use it instead of recalculating UNITS * UNITPRICE + COMMISSION et al. every time.

I've got plenty of OFX data where the server violates this part of the spec, and have deemed that data integrity trumps potential rounding errors on SQLite.  OFX data precision doesn't run too many decimal places, and so far I haven't run into cases where rounding errors occurred upon quantization.

I'm interested in any actual errors here, and plan to have reconciliation functions which will roll forward transactions from starting balances to catch discrepancies in ending balances... but to my mind, SQLite is mostly for testing purposes.  For real financial data with high accuracy needs, use a stricter DB back end.  I am happy with postgres.

**Comment by turicas** [comment]
Thanks for the explanation! :+1:
